builder/virtualbox/common/step_attach_guest_additions.go

```diff
@@ -77,7 +77,7 @@ func (s *StepAttachGuestAdditions) Cleanup(state multistep.StateBag) {
 		"--medium", "none",
 	}
 
-	if err := driver.VBoxManage(command...); err != nil {
-		ui.Error(fmt.Sprintf("Error unregistering guest additions: %s", err))
-	}
+	// Remove the ISO. Note that this will probably fail since
```

Comment:
It would be nice if we could avoid this. We already see an error message generated by the clean up [here](https://github.com/mitchellh/packer/blob/master/builder/virtualbox/iso/step_attach_iso.go#L89-L91). This caused me some concern when I first saw it as there is no mention that it is safe to ignore. The error looks very similar to those seen in #4432.

Could we not update the state bag in step_remove_devices.go e.g. ```state.Put("guest_additions_attached", false)``` and then wrap an appropriate conditional around the code in step_attach_guest_additions.go to avoid the error?

Alternatively, would you consider an upstream addition to github.com/mitchellh/multistep that allows you to remove items from the state bag once they are no longer of any use? This would allow us to use the new method to: 

```
state.Remove("guest_additions_attached")
``` 

within step_remove_devices.go and use the same conditional logic within step_attach_guest_additions.go to avoid running the command if the guest additions have already been detached. e.g.

```
if _, ok := state.GetOk("guest_additions_attached"); ok {
...
}
```

I've already tested doing just this (with the required changes to the upstream multistep code) and it seems to work quite well. If you think this would be of any benefit let me know and I'll issue the required PR's for you to take a look at.

Cheers